src/shell/shell.cpp

```diff
@@ -202,6 +202,11 @@ uint16_t get_tick_random_number() {
 	return (uint16_t)(GetTicks() % random_uplimit);
 }
 
+// Disable PVS warning 1020 for this function.
+// It's a false positive about needing to call DOS_CloseFile()
+#pragma pvs(push)
+#pragma pvs(disable: 1020)
```

Comment:
I'd suggest disabling the warning only for that single line with putting a `//-V1020` comment at the end of it.

Just run `git grep '//-V'` on our repo, we're using that form of suppression in a few places.

Reply:
That's the way I usually like to do it but it's throwing 4 separate instances of that error on different lines and on lines that only contain a closing bracket.

![pvs](https://github.com/dosbox-staging/dosbox-staging/assets/8128047/3ec4637e-e324-48d7-8a49-16d84a04215b)

Reply:
Yeah ok, if you can't refactor it somehow so the warning appears on a single line only, then the pragma approach is the better way, indeed.